cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -313,20 +312,18 @@ func performEtcdStaticPodUpgrade(client clientset.Interface, waiter apiclient.Wa
 	}
 
 	// Waiter configurations for checking etcd status
+	// If we are upgrading TLS we need to wait for old static pod to be removed.
```

Comment:
The case if we are upgrading TLS doesn't exist anymore so pod restart delay IMO should go away